gluten-core/src/main/resources/substrait/proto/substrait/algebra.proto

```diff
@@ -495,6 +504,7 @@ message Rel {
     GenerateRel generate = 17;
     WriteRel write = 18;
     TopNRel top_n = 19;
+    WindowGroupLimitRel windowGroupLimit = 20;
```

Comment:
Substrait already has a field #20.  This means that your Substrait is not going to be compatible with any other Substrait consumers or tools.  The Substrait project is open to adding new relations -- all it takes is a PR, issue, or email to start the discussion.

Reply:
@EpsilonPrime can you explain what you mean and if it's still an issue? Does any change to algebra.proto need to be updated in the Substrait project too?

Reply:
Right now the this copy differs in about a half dozen ways from the original project.  A protobuf saved using this version will be loaded incorrectly since DdlRel has field number 20.  What should happen is that the differences introduced here get applied back to the main project.  I've started the process of the CSV text format there.  The Substrait project introduced ConsistentPartitionWindowRel a while back (as field number 17) which may actually do what you want here.

As long as you only talk to other consumers using this version of Substrait your code will work.  But you're missing out on the other tools.  For instance, the Substrait Validator is great for checking that you've constructed a conforming plan.  I run all of the plans generated by my end to end tests through it and catches issues all the time.

Reply:
Got it, I'm working on a PR that also touches algebra.proto - https://github.com/apache/incubator-gluten/pull/5632/files#diff-632cd96e81af6938b453cb8ec69d66123284f17c6e47f3f16b8f08e8046afc39. Can the reconciliation happen in one shot? I can try to port these changes over with your guidance!